app/src/main/java/com/github/fribourgsdp/radio/User.kt

```diff
@@ -1,33 +1,31 @@
 package com.github.fribourgsdp.radio
 
 import android.graphics.Color
-import java.lang.Exception
-import java.lang.IllegalArgumentException
 import kotlin.random.Random
+import kotlinx.serialization.Serializable
 
-class User (nameInput: String) {
-    var name: String = nameInput
-        set(value) {
-            nameSanitizer(value)
-            field = value
-        }
+@Serializable
+class User (val name: String, val color: Int) {
     init {
-        nameSanitizer(name)
+        require(name.isNotEmpty() && name.isNotBlank())
     }
+
+    constructor(name: String): this(name, generateColor())
+
     private val playlists = mutableSetOf<Playlist>()
-    val color : Int = (255 shl 24) or
-        (Random.nextInt(100, 200) shl 16) or
-        (Random.nextInt(100, 200) shl 8) or
-        Random.nextInt(100, 200)
     private var linkedSpotify: Boolean = false
     val initial get(): Char = name.elementAt(0)
     val spotifyLinked get(): Boolean = linkedSpotify
 
-
-    private fun nameSanitizer(name: String) {
-        if (name.isEmpty() || name.isBlank()) {
-            throw  IllegalArgumentException("name cannot be empty or blank string")
+    companion object {
+        fun load(path: String) : User {
```

Comment:
When you say path, do you mean the User IDs in Firebase or something else?

Reply:
no, this is strictly for saving data locally on the device, so it is the relative path for the app specific storage

Reply:
this will become useful in the save_user branch but it has been split into two PRs for clarity